leasing/models/lease.py

```diff
@@ -51,5 +51,24 @@ def get_identifier(self):
         '''
         return '{}{}{}-{}'.format(self.type, self.municipality, self.district, self.sequence)
 
+    @staticmethod
+    def get_from_identifier(identifier):
+        parts = Lease.get_identifier_parts(identifier)
+        return Lease.objects.get(**parts)
+
+    @staticmethod
+    def get_identifier_parts(identifier):
```

Comment:
This could validate the identifier or just use regexp to parse the identifier and fail if the expression doesn't match